src/algebra/module_regular.lean

```diff
@@ -0,0 +1,179 @@
+/-
+Copyright (c) 2021 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa
+-/
+import algebra.smul_with_zero
+import algebra.regular
+/-!
+# Action of regular elements on a module
+
+We introduce `M`-regular elements, in the context of an `R`-module `M`.
+
+By definition, an `M`-regular element of a commutative ring acting on an `R`-module `M` is an
+element `a ∈ R` such that the function `m ↦ a • m` is injective.
+-/
+
+--should it be in the module namespace?
+namespace module
+
+variables {R S : Type*} {M : Type*} {a b : R} {s : S}
+
+section has_scalar
+
+variables [has_scalar R M] [has_scalar R S] [has_scalar S M] [is_scalar_tower R S M]
+
+variables (M)
+
+/-- An `M`-regular element is an element `c` such that multiplication on the left by `c` is an
+injective map `M → M`. -/
+def is_regular (c : R) := function.injective ((•) c : M → M)
```

Comment:
How about `is_smul_regular` instead of `module.is_regular`?

Reply:
I like this suggestion and I changed the name!